cylc/uiserver/data_store_mgr.py

```diff
@@ -271,16 +268,6 @@ def _update_workflow_data(self, topic, delta, w_id):
             w_id (str): Workflow external ID.
 
         """
-        # Wait until data-store is populated for this workflow,
-        # carry on otherwise, errors will be reconciled with data validation.
-        if self.data[w_id][WORKFLOW].last_updated == 0:
-            loop_cnt = 0
-            while loop_cnt < self.INIT_DATA_WAIT_CHECKS:
```

Comment:
Hold on.. I just increased this check time recently, because waiting a shorter time was causing problems for larger workflows..

Are you sure we can just delete it?

Reply:
https://github.com/cylc/cylc-uiserver/pull/721

That contains two fixes:
- increase the time waiting for the initial dump
- make sure deltas get applied even if the store doesn't have a previous update time.

So what it will do is allow any deltas to come in and get applied, even if the initial dump hasn't been applied.. Which could mean they get overwritten by the dump (so we miss deltas)

Reply:
Sorry I was planning to tag you and ask about this. I have observed in testing a workflow started via the GUI that this `while` loop is blocking any deltas coming in for the maximum wait time and removing it leads to the tasks showing much faster in the GUI. I tried increasing `INIT_DATA_WAIT_CHECKS` to 120 so that it would wait 60s and it still does not receive what it expects. Removing it did not seem to cause any problems that I observed, and did not bring back the bug reported in #700 while running the cylc-flow integration tests at `-n 8` parallelisation.

Maybe there is some other fix to make this `while` loop work as intended? But at present it does not seem to be working correctly.

Reply:
fix up:
https://github.com/cylc/cylc-uiserver/pull/761

fixes this and the other long standing problem